src/app/shared/services/ingest.service.ts

```diff
@@ -225,49 +226,48 @@ export class IngestService {
       .pipe(map(data => data._embedded && data._embedded.projects ? data._embedded.projects[0] : null));
   }
 
-  public fetchSubmissionData(submissionId, entityType, params): Observable<PagedData<MetadataDocument>> {
-    let url = `${this.API_URL}/submissionEnvelopes/${submissionId}/${entityType}`;
-    const submission_url = `${this.API_URL}/submissionEnvelopes/${submissionId}`;
+  public fetchSubmissionData(options: FetchSubmissionDataOptions): Observable<PagedData<MetadataDocument>> {
```

Comment:
This method is a bit too long. Would be nice to extract some of the logic to well named methods.